sycl/include/CL/sycl/intel/fpga_device_selector.hpp

```diff
@@ -0,0 +1,51 @@
+//==- fpga_device_selector.hpp --- SYCL FPGA device selector shortcut  -----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl.hpp>
+
+namespace cl {
+namespace sycl {
+namespace intel {
+
+class platform_selector : public default_selector {
+private:
+  std::string device_platform_name;
+
+public:
+  platform_selector(const std::string &platform_name)
+      : device_platform_name(platform_name){};
+
+  virtual int operator()(const device &device) const {
+    const platform &pf = device.get_platform();
+    const std::string &platform_name = pf.get_info<info::platform::name>();
+    if (platform_name.compare(device_platform_name) == 0) {
```

Comment:
Do we really need compare here? 
```suggestion
    if (platform_name == device_platform_name) {
```

Reply:
no. Good catch!